src/uucore/src/lib/features/checksum.rs

```diff
@@ -321,10 +322,10 @@ fn determine_regex(lines: &[String]) -> Option<(Regex, bool)> {
 
 // Converts bytes to a hexadecimal string
 fn bytes_to_hex(bytes: &[u8]) -> String {
-    bytes
-        .iter()
-        .map(|byte| format!("{byte:02x}"))
-        .collect::<String>()
+    bytes.iter().fold(String::new(), |mut bytes, byte| {
```

Comment:
Maybe create the string with `String::with_capacity(bytes.len() * 2)` first to ensure that there are no reallocations. Reallocations will typically appear as soon as there are more than 4 input bytes to format as the first allocated capacity for the string is usually 8 bytes.

Reply:
Did it.